k2/csrc/context.h

```diff
@@ -550,15 +550,51 @@ class ParallelRunner {
   // will call CudaStreamOverride::OverrideStream() and replace it
   // with this stream automatically.
   cudaStream_t NewStream() {
-    // TODO
-    return kCudaStreamInvalid;
+    DeviceType d = c_->GetDeviceType();
```

Comment:
not urgent, but I'd prefer this to be implemented in a .cu file not the header, for readability.

Reply:
Moved to `context.cu`